### PR TITLE
spotify

### PR DESCRIPTION
**Pull request by eduberenguer**

### CODE REVIEW

File: staff/edu-berenguer/spotify-app/scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
+// my custom components
```
Comment (pausanchez) [inline staff/edu-berenguer/spotify-app/scripts/main.js]: missing 'use strict'

File: staff/edu-berenguer/spotify-app/scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                return {
+                    id: obj.id,
+                    text: obj.name
+                };
+            }));
+        });
+});
+
+var resultsTrack = new ResultsList();
+var DEFAULT_IMAGE = "https://www.google.es";
```
Comment (pausanchez) [inline staff/edu-berenguer/spotify-app/scripts/main.js]: "https://www.google.es" is not an image url

File: staff/edu-berenguer/spotify-app/scripts/logic.spec.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 'use strict';
```
Comment (bcasasus) [inline staff/edu-berenguer/spotify-app/scripts/logic.spec.js]: well done using 'use strict'

File: staff/edu-berenguer/restaurants-app/scripts/main.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
+// my custom components
```
Comment (bcasasus) [inline staff/edu-berenguer/restaurants-app/scripts/main.js:1]: 'use strict' required!!!!

File: staff/edu-berenguer/restaurants-app/scripts/main.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+SearchPanel.prototype = Object.create(Component.prototype);
+SearchPanel.prototype.constructor = SearchPanel;
+
+SearchPanel.prototype.onSearch = function (callback) {
+    this._callback = callback;
+};
+
+function ResultsList() {
```
Comment (bcasasus) [inline staff/edu-berenguer/restaurants-app/scripts/main.js]: maybe you could use `/***` to comment the function of the methods

File: staff/edu-berenguer/restaurants-app/scripts/main.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+var results = new ResultsList();
+
+document.body.appendChild(search.element);
+document.body.appendChild(results.element);
+
+// second search
+
+var search2 = new SearchPanel();
+
+search2.onSearch(function (query) {
```
Comment (bcasasus) [inline staff/edu-berenguer/restaurants-app/scripts/main.js]: the name search2 is not semmantic, and the functionality is redundant

File: staff/edu-berenguer/spotify-app/scripts/logic.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 var logic = {
-    token: 'BQCJBxeav8UJ-UQrfRWo28ibBCm3byJ1vgpXxCgRUPnF3Rp6AcgiZ3IdsS_0AQ2mMGyJDXGQMHEBseN8XtzYSdKKQNb-ONbjxSHWnIvTFhMem7Tx2FJVnrASA_EsPHWVfl9tzF7WpkLsSFmpYzc',
+    token: 'BQBCqt-fBKY1U80-xWs2gbJ7THSp0Cpx2RVu8s2I-od8U9VLxGpd6J4gIzsEUS1huKMCmCpThYVxMWyCckdKg9dkj-NO-GckM6KmzBX-hE3bCD4ko5mk4tvDIHjD2K-L_PfkGHrDB-tI1VMeOyU',
 
     _callApi: function (path) {
```
Comment (bcasasus) [inline staff/edu-berenguer/spotify-app/scripts/logic.js]: Maybe you could use `  _callApi: function (path) {
        return $.ajax('https://api.spotify.com/v1' + path, {
            headers: {
                authorization: 'Bearer ' + this.token
            }
        })
        .catch(function(err) {
            throw Error('request error, status ' + err.status);
        });
    }`

File: staff/edu-berenguer/spotify-app/style.css [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    /* background-color: red; */
+    width: 70% !important;
+    background-color: transparent !important;
+}
+
+body {
+    background: #22d243;  /* fallback for old browsers */
+    background: -webkit-linear-gradient(to right, #08EA90, #22d243);  /* Chrome 10-25, Safari 5.1-6 */
+    background: linear-gradient(to right, #08EA90, #22d243); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */    
+    background: linear-gradient(to right, #08EA90, #22d243); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
```
Comment (pausanchez) [inline staff/edu-berenguer/spotify-app/style.css]: Estilo doble

File: staff/edu-berenguer/array-native/scripts/copy-with-in.test.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        // expected output: [4, 2, 3, 4, 5]
+
+        //put at position 1 the values after position 3
+        var result2 = array2.copyWithin(1, 3);
+        // expected output: [1, 4, 5, 4, 5]
+        
+        expect(result).toEqual([4, 2, 3, 4, 5]);
+        expect(result.length).toBe(5);
+
+        expect(result2).toEqual([1, 4, 5, 4, 5])
```
Comment (SergioCastilloSorribes) [inline staff/edu-berenguer/array-native/scripts/copy-with-in.test.js:24]: Falta semicolon!!!! Esto no es react!!!Tienes que mejorar!!!

**Review by SergioCastilloSorribes — CHANGES_REQUESTED** [review]

**Pull request merged**